Source/Fuse.Maps/MapTile.uno

```diff
@@ -0,0 +1,298 @@
+
+				var i = new Fuse.Controls.Image()
+				{
+					File = fs,
+					Width = 10,
+					X = p.X,
+					Y = p.Y
+				};
+				var _trans = new Fuse.Translation()
+				{
+					RelativeNode = i,
+					RelativeTo   = Fuse.TranslationModes.Size,
+					X = m.IconAnchorX - 0.5f,
+					Y = m.IconAnchorY * -1
+				};
+
+				Children.Add(i);
+				BringToFront(i);
+				i.Children.Add(_trans);
+				_markers.Add(i);
+			}
+		}
+
+		bool ready = false;
+		protected override void OnRooted()
+		{
+			base.OnRooted();
+			MapView m = Parent as MapView;
+			m.ClipToBounds = true;
+			ready = true;
+		}
+
+		float _grid_size = 0;
+		public void OnPlaced(object sender, PlacedArgs args)
+		{
+			var s = Math.Min(ActualSize.X, ActualSize.Y);
+			grid.Width = s;
+			grid.Height = s;
+			_grid_size = s;
+			UpdateMap();
+		}
+
+		double _lng = 0;
+		public double Longitude
+		{
+			get
+			{
+				MapView m = Parent as MapView;
+				if (m != null)
+				{
+					_lng = m.Longitude; 
+				}
+				return _lng;
+			}
+			set
+			{
+				_lng = value;
+				MapView m = Parent as MapView;
+				if (m != null)
+				{
+					m.Longitude = value; 
+				}
+				else
+				{
+					UpdateMap();
+				}
+			}
+		}
+
+		double _lat = 0;
+		public double Latitude
+		{
+			get
+			{
+				MapView m = Parent as MapView;
+				if (m != null)
+				{
+					_lat = m.Latitude; 
+				}
+				return _lat;
+			}
+			set
+			{
+				_lat = value;
+				MapView m = Parent as MapView;
+				if (m != null)
+				{
+					m.Latitude = value; 
+				}
+				else
+				{
+					UpdateMap();
+				}
+			}
+		}
+
+		double _zoom = 2;
+		public double Zoom
+		{
+			get
+			{
+				MapView m = Parent as MapView;
+				if (m != null)
+				{
+					_zoom = m.Zoom; 
+				}
+				return _zoom;
+			}
+			set
+			{
+				_zoom = value;
+				MapView m = Parent as MapView;
+				if (m != null)
+				{
+					m.Zoom = value; 
+				}
+				else
+				{
+					UpdateMap();
+				}
+			}
+		}
+
+		float2 _tilepos = float2(0);
+		float2 _cornertile = int2(0);
+		public void UpdateMap()
+		{
+			MapView m = Parent as MapView;
+			if (m == null) return;
+			var p = WorldToTilePos(Longitude, Latitude, (int)Zoom);
+			_tilepos = p;
+			var i = 0;
+			var tmp = Math.Floor(p);
+			_cornertile = (int2)tmp - 1; // Have the corner one to the left and above the center of the map
+
+			// Calculate offset (adjustment) of tile, compared to center of map:
+			var adj = float2(0.5f) - (p - tmp);
+			adj = adj * maps[4].ActualSize;
+			trans.XY = adj;
+
+			for (var y = (int)p.Y - 1; y < (int)p.Y + 2; y++)
+			{
+				for (var x = (int)p.X - 1; x < (int)p.X + 2; x++)
+				{
+					maps[i].Url = MakeUrl((int)Zoom, x, y);
+					i++;
+				}
+			}
+			UpdateMarkers();
+		}
+
+		public float2 WorldToTilePos(double lon, double lat, int zoom)
+		{
+			float2 p = float2(0);
+			p.X = (float)((lon + 180.0) / 360.0 * (1 << zoom));
+			p.Y = (float)((1.0 - Math.Log(Math.Tan(lat * Math.PI / 180.0) + 
+				1.0 / Math.Cos(lat * Math.PI / 180.0)) / Math.PI) / 2.0 * (1 << zoom));
```

Comment:
Please use `Math.Pow(2,zoom)` instead of `1 << zoom` when using floats.
Use the `Math.DegreessToRadians` and RadiansToDegrees functions to clarify this code.